components/RegionDialog/RegionDialog.tsx

```diff
@@ -0,0 +1,104 @@
+import { useRouter } from "next/router";
+import React from "react";
+import { useForm } from "react-hook-form";
+
+import { clearCheckout } from "@/lib/checkout";
+import { DEFAULT_LOCALE, LOCALES } from "@/lib/regions";
+
+import { Button } from "../Button";
+import useChannels from "../ChannelsProvider/useChannels";
+
+interface RegionDialogProps {
+  onClose: () => void;
+  isOpen: boolean;
+}
+
+export interface RegionFormData {
+  channel: string;
+  locale: string;
+}
+
+const RegionDialog = ({ isOpen, onClose }: RegionDialogProps) => {
+  const router = useRouter();
+  const { channels, currentChannel, setCurrentChannel } = useChannels();
+
+  const { register, handleSubmit } = useForm<RegionFormData>({
+    defaultValues: {
+      channel: currentChannel.slug,
+      locale: router.locale || DEFAULT_LOCALE,
+    },
+  });
+
+  const onSubmit = handleSubmit(async (formData: RegionFormData) => {
+    if (formData.channel !== currentChannel.slug) {
+      await setCurrentChannel(formData.channel);
+      clearCheckout();
+    }
+    onClose();
+
+    // Update current URL to use the chosen channel
+    await router.push({
+      pathname: router.pathname,
+      query: {
+        ...router.query,
+        channel: formData.channel,
+        locale: formData.locale,
+      },
+    });
+  });
+
+  if (!isOpen) {
+    return null;
+  }
+
+  return (
+    <div
```

Comment:
Modals should always be wrapped inside React Portals, perhaps it'd be a good idea to use a headlessui from taildwind? https://headlessui.dev/react/dialog

Reply:
I think since css is here is not to show best practices but only to make the example easy to navigate etc. I'd just stick to the simplest solution we can get and leave it just with portal